### PR TITLE
Add LIMIT clause support

### PR DESCRIPTION
**Pull request by alex-dukhno**
Add support `LIMIT <n>` for streaming sql engine 

this is a platform version of https://github.com/hazelcast/hazelcast-jet/pull/2976

### CODE REVIEW

File: hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                new Row(4L),
+                new Row(5L),
+                new Row(6L),
+                new Row(7L),
+                new Row(8L),
+                new Row(9L))
+        );
+    }
+
+    protected void checkFailure0(
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java:161]: Could not it be simply replaced with the usage of `Assertions.assertThatThrownBy()` ?

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetQueryResultProducer.java [hunk 2]
```diff
@@ -71,6 +76,13 @@ public void consume(Inbox inbox) {
         ensureNotDone();
         for (Object[] row; (row = (Object[]) inbox.peek()) != null && rows.offer(new HeapRow(row)); ) {
             inbox.remove();
+            if (limit != Long.MAX_VALUE) {
+                limit -= 1;
+                if (limit < 1) {
+                    done.compareAndSet(null, new ResultLimitReachedException());
+                }
+                ensureNotDone();
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetQueryResultProducer.java:84]: Why is it needed to be checked on each iteration?

File: hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    @Test
+    public void negativeLimitValue() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        Assertions.assertThatThrownBy(() -> instance().getSql().execute("SELECT name FROM " + tableName + " LIMIT -10"))
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java:79]:
```suggestion
        Assertions.assertThatThrownBy(() -> sqlService.execute("SELECT name FROM " + tableName + " LIMIT -10"))
```

File: hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    private static void assertContainsOnlyOneOfRows(String sql, Collection<Row> expectedRows) {
+        assertContainsSubsetOfRows(sql, 1, expectedRows);
+    }
+
+    /**
+     * Asserts that the result of {@code sql} contains a subset of {@code expectedRows}, but
+     * only a subset of them with size of {@code subsetSize}.
+     */
+    private  static void assertContainsSubsetOfRows(String sql, int subsetSize, Collection<Row> expectedRows) {
+        SqlService sqlService = instance().getSql();
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java:169]:
```suggestion
    private static void assertContainsSubsetOfRows(String sql, int subsetSize, Collection<Row> expectedRows) {
```

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetQueryResultProducer.java [hunk 5]
```diff
@@ -71,6 +76,13 @@ public void consume(Inbox inbox) {
         ensureNotDone();
         for (Object[] row; (row = (Object[]) inbox.peek()) != null && rows.offer(new HeapRow(row)); ) {
             inbox.remove();
+            if (limit != Long.MAX_VALUE) {
+                limit -= 1;
+                if (limit < 1) {
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetQueryResultProducer.java:81]:
```suggestion
                limit--;
                if (limit == 0) {
```

**Review by viliam-durina — APPROVED** [review]

**Review by gierlachg — APPROVED** [review]

**Pull request merged**